virtual_labs/tests/projects/test_project_creation.py

```diff
@@ -121,3 +122,30 @@ async def test_vlm_project_with_members_creation(
     assert response.status_code == 200
     failed_invites = response.json()["data"]["failed_invites"]
     assert failed_invites == []
+
+
+@pytest.mark.asyncio
+async def test_aggregate_views_created_on_project_creation(
+    mock_create_project: tuple[Response, dict[str, str], dict[str, str]],
+) -> None:
+    (response, headers, payload) = mock_create_project
+
+    project_id = response.json()["data"]["project"]["id"]
+    virtual_lab_id = response.json()["data"]["project"]["virtual_lab_id"]
+
+    def views_created() -> bool:
+        nexus_views = get(
+            f"{settings.NEXUS_DELTA_URI}/views/{virtual_lab_id}/{str(project_id)}",
+            headers=get_client_headers(),
+        )
+        data = nexus_views.json()
+        agg_view = [
+            view
+            for view in data.get("_results")
+            if sorted(view.get("@type")) == sorted(AGGREGATE_ELASTIC_SEARCH_VIEW)
+        ]
+        return len(agg_view) == 1
+
+    # Nexus takes some time to return the created views that's why wait until is needed here
+    aggregate_view_created = await wait_until(views_created, 5, 0.25)
+    assert aggregate_view_created is True
```

Comment:
Instead of `wait_until` you can simply use `httpx.get` with a timeout:

```suggestion
def async views_created() -> bool:
    async with httpx.AsyncClient() as client:
        try: 
            nexus_views = client.get(
                f"{settings.NEXUS_DELTA_URI}/views/{virtual_lab_id}/{str(project_id)}",
                headers=get_client_headers(),
                timeout=5
            )
        except httpx.TimeoutException:
            return False
        
        data = nexus_views.json()
        
        agg_view = [
            view
            for view in data.get("_results")
            if sorted(view.get("@type")) == sorted(AGGREGATE_ELASTIC_SEARCH_VIEW)
        ]
        
        return len(agg_view) == 1

    
    aggregate_view_created = await views_created()
    assert aggregate_view_created is True
```

Reply:
:+1: